orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/SimpleTask.java

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.api.SimpleStage;
+import com.netflix.spinnaker.orca.api.SimpleStageInput;
+import com.netflix.spinnaker.orca.api.SimpleStageOutput;
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.GenericTypeResolver;
+import org.springframework.core.ResolvableType;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class SimpleTask implements Task {
+  private SimpleStage simpleStage;
+
+  SimpleTask(@Nullable SimpleStage simpleStage) {
+    this.simpleStage = simpleStage;
+  }
+
+  private SimpleStageInput getStageInput(Stage stage) {
+    ObjectMapper objectMapper = OrcaObjectMapper.newInstance();
+
+    try {
+      List<Class<?>> cArg = Arrays.asList(SimpleStageInput.class);
+      Method method = simpleStage.getClass().getMethod("execute", cArg.toArray(new Class[0]));
+      Type inputType = ResolvableType.forMethodParameter(method, 0).getGeneric().getType();
+      Map<TypeVariable, Type> typeVariableMap =
+          GenericTypeResolver.getTypeVariableMap(simpleStage.getClass());
+
+      return new SimpleStageInput(
+          objectMapper.convertValue(
+              stage.getContext(), GenericTypeResolver.resolveType(inputType, typeVariableMap)));
+    } catch (NoSuchMethodException exeception) {
+      log.error("Cannot get method for " + simpleStage.getName() + ": " + exeception.getMessage());
+    }
+
+    return new SimpleStageInput(new Object());
+  }
+
+  @Nonnull
+  public TaskResult execute(@Nonnull Stage stage) {
+    ExecutionStatus status;
+    SimpleStageInput simpleStageInput = getStageInput(stage);
+    SimpleStageOutput output = simpleStage.execute(simpleStageInput);
+
+    switch (output.getStatus()) {
+      case TERMINAL:
+        status = ExecutionStatus.TERMINAL;
+        break;
+      case RUNNING:
+        status = ExecutionStatus.RUNNING;
+        break;
+      case COMPLETED:
+        status = ExecutionStatus.SUCCEEDED;
+        break;
+      case NOT_STARTED:
+        status = ExecutionStatus.NOT_STARTED;
+        break;
+      default:
+        status = ExecutionStatus.FAILED_CONTINUE;
+        break;
+    }
+
+    return TaskResult.builder(status)
+        .context(output.getOutputs())
```

Comment:
This doesn't seem quite what we want ... we need to be careful about what gets made available to downstream stages vs added back into the local stage context.

Reply:
This was my fault. I didn't think that one through. You're right, I don't think we want to dump it back into the context. +1.

Reply:
@ethanfrogers @ajordens do we want to git rid of `SimpleStageOutput` then and have the `execute` method return a `SimpleStageStatus` instead? That would also solve the issue above about making `SimpleStageOutput` use a generic

Reply:
I don't think so. I think `SimpleStageOutput` is fine. I def want to be able to output things from a stage so that it lands back in the overall pipeline context. I think @ajordens point here is that we might not want the things meant for output to effect the context all the time. What if we added a `getContext` method alongside the `getOutput` method. That way the stage could supply both explicitly.

Reply:
Yes to @ethanfrogers, we definitely want a task to be able to return values ... just have some control around how they're exposed (or not) to downstream stages.